File: Reverse Prefix of Word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        i = word.find(ch)
        if i == -1:
            return word
        return word[i::-1] + word[i+1:]
# class Solution:
#     def reversePrefix(self, word: str, ch: str) -> str:
#         stack = []
#         for i, c in enumerate(word):
#             stack.append(c)
#             if ch == c:
#                 return "".join(stack[::-1]) + word[i+1:]
#         return word

File: Reverse Prefix of Word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int i = word.find(ch);
        if (-1 == i) return word;

        stringstream ss;
        for (int j = i; j >= 0; j--) {
            ss << word[j];
        }
        ss << word.substr(i+1);
        return ss.str();
    }
};

